frontend/src/pages/modelServing/screens/projects/ModelServingPlatform.tsx

```diff
@@ -44,6 +49,22 @@ const ModelServingPlatform: React.FC = () => {
 
   const servingPlatformStatuses = useServingPlatformStatuses();
 
+  const isNIMModelServingAvailable = useIsAreaAvailable(SupportedArea.NIM_MODEL).status;
+  const [isNIMAPIKeyValid, setIsNIMAPIKeyValid] = useState<boolean>(false);
+  const dashboardNamespace = useDashboardNamespace();
+
+  useEffect(() => {
+    const checkAPIKey = async () => {
+      try {
+        const valid = await isNIMAPIKeyEnabled(dashboardNamespace);
+        setIsNIMAPIKeyValid(valid);
+      } catch (error) {
+        setIsNIMAPIKeyValid(false);
+      }
+    };
+    checkAPIKey();
+  }, [dashboardNamespace]);
```

Comment:
Alright, this is a NIT -- we may need to swing back for this as we are trying to do this today.

Please make this a custom hook so logic is repeatable and reusable.